js/script.js

```diff
@@ -313,43 +322,52 @@ const eventHandlers = {
 const apiService = {
   async fetchMovies() {
     try {
+      // First check if data exists in localStorage
```

Comment:
Note that this paradigm of returning the cache first comes with the drawback of displaying outdated data. In your case, I doubt you will often update the JSON file you use as a database. However, it is something to keep in mind when you work on future projects.

Another thing to consider is the amount of data you are storing. The local storage has a size limitation of ~5MB. I am unsure whether it is shared across applications. But the thing I am sure of is that your users won't be happy if you fill up their devices with data 😉 

In any case, **you don't need to change anything** now. It is cool that you experimented with local storage. I just wanted to give you some brain food for the future.